modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientInboundMessageHandler.java

```diff
@@ -460,17 +465,22 @@ private void handshake(ChannelHandlerContext ctx, ClientMessageUnpacker unpacker
                 return;
             }
 
-            authenticationManager
-                    .authenticateAsync(createAuthenticationRequest(clientHandshakeExtensions))
-                    .handleAsync((user, err) -> {
-                        if (err != null) {
-                            handshakeError(ctx, packer, err);
-                        } else {
-                            handshakeSuccess(ctx, packer, user, clientFeatures, clientVer, clientCode);
-                        }
+            handshakeEventLoopSwitcher.switchEventLoopIfNeeded(channelHandlerContext.channel(),
+                            () -> authenticationManager.authenticateAsync(createAuthenticationRequest(clientHandshakeExtensions))
+                                    .handleAsync((user, err) -> {
+                                        if (err != null) {
+                                            handshakeError(ctx, packer, err);
+                                        } else {
+                                            handshakeSuccess(ctx, packer, user, clientFeatures, clientVer, clientCode);
+                                        }
+
+                                        return null;
+                                    }, ctx.executor()))
+                    .exceptionally(t -> {
```

Comment:
I wonder why this `exceptionally()` is added

Reply:
It is necessary to release context (avoid Netty leak) when an exception happens on  the switching event loop.